docs/cert-manager.md

```diff
@@ -1,76 +1,448 @@
+
+```shell
+sts
+sts.minio-operator.svc.
+sts.minio-operator.svc.<cluster domain>
+```
+
+> [!IMPORTANT]
+> Replace `<cluster domain>` with the actual values for your MinIO tenant.
+> `cluster domain` is the internal root DNS domain assigned in your Kubernetes cluster. Typically this is `cluster.local`, check on your coredns
+> configuration for the correct value for your Kubernetes cluster. For example, using `kubectl get configmap coredns -n kube-system -oyaml | yq ".data"`. 
+> The way the root DNS domain is managed can vary depending on the Kubernetes distribution (Openshift, Rancher, EKS, etc.)
+
+Create a `Certificate` for the domains mentioned above:
+
+```yaml
+# sts-tls-certificate.yaml
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: sts-certmanager-cert
+  namespace: minio-operator
+spec:
+  dnsNames:
+    - sts
+    - sts.minio-operator.svc
+    - sts.minio-operator.svc.cluster.local
+  secretName: sts-tls
+  issuerRef:
+    name: minio-operator-ca-issuer
+```
+
+```shell
+kubectl apply -f sts-tls-certificate.yaml
 ```
+This creates a secret called `sts-tls` in the `minio-operator` namespace.
 
-This file request `cert-manager` to issue a new certificate based on the following internal domains.
+> [!IMPORTANT]
+> The secret name is not optional. Make sure the secret name is `sts-tls` by setting `spec.secretName: sts-tls` as in the example above.
+
+### Create TLS certificate for Console service
+
+Request Cert Manager to issue a new certificate including the following DNS domains for console:
+
+```shell
+console
+console.minio-operator.svc.
+console.minio-operator.svc.<cluster domain>
+```
+
+> [!IMPORTANT]
+> Replace `<cluster domain>` with the actual values for your MinIO tenant.
+> `cluster domain` is the internal root DNS domain assigned in your Kubernetes cluster. Typically this is `cluster.local`, check on your coredns
+> configuration for the correct value for your Kubernetes cluster. For example, using `kubectl get configmap coredns -n kube-system -oyaml | yq ".data"`.
+> The way the root DNS domain is managed can vary depending on the Kubernetes distribution (Openshift, Rancher, EKS, etc.)
+
+Create a `Certificate` for the domains mentioned above:
+
+```yaml
+# console-tls-certificate.yaml
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: console-certmanager-cert
+  namespace: minio-operator
+spec:
+  dnsNames:
+    - console
+    - console.minio-operator.svc
+    - console.minio-operator.svc.cluster.local
+  secretName: console-tls
+  issuerRef:
+    name: minio-operator-ca-issuer
+```
+
+```shell
+kubectl apply -f console-tls-certificate.yaml
+```
+
+This creates a secret called `console-tls` in the `minio-operator` namespace.
+
+> [!IMPORTANT]
+> The secret name is not optional. Make sure the secret name is `console-tls` by setting `spec.secretName: console-tls` as in the example above.
+
+### Install Operator with Auto TLS disabled for STS
```

Comment:
Perhaps suggest:
```
kubectl patch deployment -n minio-operator minio-operator -p '{"spec":{"template":{"spec":{"containers":[{"name": "minio-operator", "env": [{"name": "OPERATOR_STS_AUTO_TLS_ENABLED", "value": "off"}, {"name": "MINIO_CONSOLE_TLS_ENABLE", "value": "off"}]}]}}}}'
```

Reply:
a patch is not neccesary, the example yaml above already have this secret name, here I just wanted to highlight that what the secret name should be and where should set it.

Reply:
at the moment Operator deployment is created should have already the values, otherwise have adverse effects, I am adding instructions to do it with a kustomization

```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

resources:
- github.com/minio/operator/resources

patches:
- patch: |-
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: minio-operator
      namespace: minio-operator
    spec:
      template:
        spec:
          containers:
            - name: minio-operator
              env:
                - name: MINIO_CONSOLE_TLS_ENABLE
                  value: "off"
                - name: OPERATOR_STS_AUTO_TLS_ENABLED
                  value: "off"
                - name: OPERATOR_STS_ENABLED
                  value: "on"

```